devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1255,109 +1408,169 @@ void YarpRobotLoggerDevice::run()
     const double time = std::chrono::duration<double>(BipedalLocomotion::clock().now()).count();
 
     std::lock_guard lock(m_bufferManagerMutex);
-    // collect the data
+
+    Eigen::Matrix<double, 1, 1> timeData;
+    timeData << time;
+
+    std::string signalName = "";
+    std::string rtSignalName = "";
+
+    rtSignalName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(TIMESTAMPS_NAME);
+    vectorCollectionRTDataServer.populateData(rtSignalName, timeData);
+
     if (m_streamJointStates)
     {
         if (m_robotSensorBridge->getJointPositions(m_jointSensorBuffer))
         {
-            m_bufferManager.push_back(m_jointSensorBuffer, time, "joints_state::positions");
+            m_bufferManager.push_back(m_jointSensorBuffer, time, JOINT_STATE_POSITIONS_NAME);
+            rtSignalName = std::string(ROBOT_RT_ROOT_NAME) + std::string(TREE_DELIM) + std::string(JOINT_STATE_POSITIONS_NAME);
+            vectorCollectionRTDataServer.populateData(rtSignalName, m_jointSensorBuffer);
```

Comment:
Same as above. I think you can condense these operations into a function.

Reply:
Fixed in [834d618](https://github.com/ami-iit/bipedal-locomotion-framework/commit/834d6186c3c2cf35457bbae386c0d8d627ff69e9)